Public/Src/Engine/Processes/ProcessTimes.cs

```diff
@@ -124,5 +125,17 @@ public DateTime ExitTimeUtc
         /// Total time between creation and exit.
         /// </summary>
         public TimeSpan TotalWallClockTime => StartTimeUtc < ExitTimeUtc ? ExitTimeUtc - StartTimeUtc : TimeSpan.Zero;
+
+        /// <nodoc />
+        public void Serialize(BinaryWriter writer)
+        {
+            writer.Write(m_create);
+            writer.Write(m_exit);
+            writer.Write(m_kernel);
+            writer.Write(m_user);
+        }
+
+        /// <nodoc />
+        public static ProcessTimes Deserialize(BinaryReader reader) => new ProcessTimes(reader.ReadInt64(), reader.ReadInt64(), reader.ReadInt64(), reader.ReadInt64());

```

Comment:
each on their own line with a named argument will make it easier to maintain parity bewteen read & write long-term.

Reply:
```
public static ProcessTimes Deserialize(BinaryReader reader) => new ProcessTimes(
    create: reader.ReadInt64(), 
    exit: reader.ReadInt64(), 
    kernel: reader.ReadInt64(), 
    user: reader.ReadInt64()
);
```

---
In reply to: [276815695](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276815695) [](ancestors = 276815695)